packages/quantic/force-app/main/default/lwc/quanticRecentQueries/quanticRecentQueries.js

```diff
@@ -0,0 +1,68 @@
+import { LightningElement, api, track } from 'lwc';
+import { registerComponentForInit, initializeWithHeadless } from 'c/quanticHeadlessLoader';
+import { getItemfromLocalStorage, setIteminLocalStorage } from 'c/quanticUtils';
+
+export default class QuanticRecentQueries extends LightningElement {
+  /** @type {string} */
+  localStorageKey = 'quantic-recent-queries';
+  /** @type {import("coveo").RecentQueriesState} */
+  @track state = {
+    queries: getItemfromLocalStorage(this.localStorageKey) ?? [],
+    maxLength: 10,
+  };
+  /** @type {import("coveo").RecentQueriesList} */
+  recentQueriesList;
+  /** @type {()=> void} */
+  unsubscribe;
+  /** @type {boolean} */
+  isCollapsed = false;
+  /** @type {string} */
+  collapseIcon = 'utility:dash';
+  /** @type {string} */
+  @api label;
+  /** @type {string} */
+  @api engineId;
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize.bind(this));
+  }
+
+  /**
+  * @param {import("coveo").SearchEngine} engine
+  */
+  @api
+  initialize(engine) {
+    this.recentQueriesList = CoveoHeadless.buildRecentQueriesList(engine, {
+      initialState: {
+        queries: this.state.queries,
+      },
+      options: {
+        maxLength: this.state.maxLength,
+      },
```

Comment:
```suggestion
    this.recentQueriesList = CoveoHeadless.buildRecentQueriesList(engine, {
      initialState: {
        queries:  getItemfromLocalStorage(this.localStorageKey) ?? [],
      },
      options: {
        maxLength: this.maxLength,
      },
```

Reply:
This might even be exactly what you had before. in which case i apologize 😛

Reply:
haha yes. nvm i'll change this again :joy: